File: AVS_Second_Task/shape.cpp
#include "triangle.h"
#include "circle.h"
#include "rectangle.h"

/**
 * Вывод цвета в файл.
 * @param outFile - файл для вывода.
 */
void Shape::WriteColorInFile(FILE *outFile) {
    switch (color) {
        case RED:
            fprintf(outFile, "\tЦвет: Красный\n");
            break;
        case ORANGE:
            fprintf(outFile, "\tЦвет: Оранжевый\n");
            break;
        case YELLOW:
            fprintf(outFile, "\tЦвет: Желтый\n");
            break;
        case GREEN:
            fprintf(outFile, "\tЦвет: Зеленый\n");
            break;
        case LIGHTBLUE:
            fprintf(outFile, "\tЦвет: Голубой\n");
            break;
        case BLUE:
            fprintf(outFile, "\tЦвет: Синий\n");
            break;
        case PURPLE:
            fprintf(outFile, "\tЦвет: Фиолетовый\n");
            break;
    }
}

/**
 * Геттер для цвета.
 * @return цвет.
 */
Colors Shape::GetColor() {
    return color;
}

/**
 * Сеттер для цвета.
 * @param color - цвет.
 */
void Shape::SetColor(Colors newColor) {
    color = newColor;
}

/**
 * Статический ввод фигуры.
 * @param inputFile - входной файл.
 * @return экзмепляр фигуры.
 */
Shape *Shape::StaticIn(FILE *inputFile) {
    int k;
    bool check = Exceptions::GetInt(inputFile, &k);

    int colorType;
    bool checkColor = Exceptions::GetInt(inputFile, &colorType);
    if (!check || !checkColor)
        return nullptr;
    Shape *shape;
    switch (k) {
        case 1:
            shape = new Rectangle;
            break;
        case 2:
            shape = new Triangle;
            break;
        default:
            shape = new Circle;
            break;
    }
    shape->SetColor(Colors(colorType));
    shape->In(inputFile);
    return shape;
}

/**
 * Статический метод генерации рандомной фигуры.
 * @return фигура.
 */
Shape *Shape::StaticInRnd() {
    auto k = Random::next(1, 4);
    Shape *shape;
    switch (k) {
        case 1:
            shape = new Rectangle;
            break;
        case 2:
            shape = new Triangle;
            break;
        default:
            shape = new Circle;
            break;
    }
    shape->SetColor(Colors(Random::next(0, 7)));
    shape->InRnd();
    return shape;
}

File: avs_third_task/shape.py
from enum import Enum

# Перечислимый тип цветов фигуры.
class Colors(Enum):
    TRANSPARENT = 0
    RED = 1
    ORANGE = 2
    YELLOW = 3
    GREEN = 4
    LIGHTBLUE = 5
    BLUE = 6
    PURPLE = 7


# Класс фигуры.
class Shape:
    def __init__(self):
        self.color = Colors.TRANSPARENT

    # Установка цвет фигуры.
    def SetColor(self, i):
        self.color = i

    # Получение цвета фигуры.
    def GetColorString(self) -> str:
        if self.color == 1:
            return "RED"
        elif self.color == 2:
            return "ORANGE"
        elif self.color == 3:
            return "YELLOW"
        elif self.color == 4:
            return "GREEN"
        elif self.color == 5:
            return "LIGHTBLUE"
        elif self.color == 6:
            return "BLUE"
        elif self.color == 7:
            return "PURPLE"
        else:
            self.color = 0
            return "TRANSPARENT"

    # Создание фигуры.
    def Read(self, strArray, i):
        pass

    # Вывод данных о фигуре в консоль.
    def Print(self):
        pass

    # Вывод данных о фигуре в файл.
    def Write(self, ostream):
        pass

    # Вычисление площади фигуры.
    def Square(self):
        pass